resources/default.rb

```diff
@@ -104,13 +107,33 @@
     action :enable
   end
 
+  pwchanger = "admin_change_password for #{new_resource.service_name}@#{port}"
+
   # Allow for Nexus to fully start before moving on.
-  ruby_block 'block until operational' do
+  ruby_block blocker do
     block do
-      Chef::Log.info "Waiting until Nexus is listening on port #{node['nexus3']['properties_variables']['port']}"
-      wait_until_ready!(node['nexus3']['api']['endpoint'], node['nexus3']['api']['wait'])
+      Chef::Log.info "Waiting until Nexus #{new_resource.service_name} is listening on port #{port}"
+      wait_until_ready!(Nexus3::Api.endpoint(port), node['nexus3']['api']['wait'])
     end
     action :nothing
+    notifies :create, "nexus3_api[#{pwchanger}]"
+    notifies :run, "nexus3_api[#{pwchanger}]"
```

Comment:
The blocker resource is notified on nexus template changes, so your `nexus3_api[pwchanger]`. Your code won't work if the password file has been removed.
I would recommend to replace delayed/immediate notification by :before ones.

Reply:
I think that the code will work as the `only_if` is taken into account.